pandas/tests/formats/test_format.py

```diff
@@ -4161,6 +4161,21 @@ def test_dates_display(self):
         self.assertEqual(result[1].strip(), "NaT")
         self.assertEqual(result[4].strip(), "2013-01-01 09:00:00.000000004")
 
+    def test_datetime64formatter(self):
+        x = Series([datetime(2016,1,1), datetime(2016,2,2), datetime(2016,3,3)])
+        def format_func(x): return x.strftime('%Y-%m')
+        formatter = fmt.Datetime64Formatter(x, formatter=format_func)
```

Comment:
add one that starts with hhmmss (and say microseconds), then formats to something else.